libraries/lib-time-and-pitch/StaffPadTimeAndPitch.cpp

```diff
@@ -56,6 +70,7 @@ void StaffPadTimeAndPitch::GetSamples(float* const* output, size_t outputLen)
       // Pass-through
       return mAudioSource.Pull(output, outputLen);
 
+   std::lock_guard<std::mutex> lock(mTimeAndPitchMutex);
```

Comment:
Oh, now I see at this commit why the member becomes non-const.

Consider using the utility `AtomicUniquePointer` instead of a mutex.

Reply:
I don't think an atomic would be helpful here, since it's not only just the reallocation of the object that must be synchronized. Remember that comment in SP:
```
  /**
    Set independent time stretch and pitch factors (synchronously to processing thread).
    ...
  */
  void setTimeStretchAndPitchFactor(double timeStretch, double pitchFactor);
  ```